lib/elasticity/index_config.rb

```diff
@@ -60,5 +63,12 @@ def validate!
         raise "#{attr} is not set" if public_send(attr).nil?
       end
     end
+
+    def settings
+      if number_of_shards.present?
+        @elasticity_config.settings[:number_of_shards] = number_of_shards
+      end
```

Comment:
I don't think you want to set `:number_of_shards` into `settings` like this.  This `@elasticity_config` was handed to you, and thus you're mutating something that _*may*_ be used in other places without you knowing, and thus cause unintended side effects.  I think you'll want to do something like:
```
def definition_settings
   return @elasticity_config.settings unless number_of_shards
    
   @elasticity_config.settings.dup.tap do | hash|
     hash[:number_of_shards] ||= number_of_shards if number_of_shards
    end
  end
```